apps/gdal_create.cpp

```diff
@@ -31,45 +31,169 @@
-        "       [-if <input_dataset>]\n"
-        "       <out_dataset>\n");
-
-    exit(bIsError ? 1 : 0);
+    GDALDestroy();
+    exit(nCode);
 }
 
 /************************************************************************/
-/*                      ArgIsNumericCreate()                            */
+/*                     GDALCreateOptions                               */
 /************************************************************************/
 
-static bool ArgIsNumericCreate(const char *pszArg)
+struct GDALCreateOptions
+{
+    int nBandCount = -1;
+    int nPixels = 0;
+    bool bPixelsSet{false};
+    int nLines = 0;
+    GDALDataType eDT = GDT_Unknown;
+    double dfULX = 0;
+    double dfULY = 0;
+    double dfLRX = 0;
+    double dfLRY = 0;
+    int nULCounter{0};
+    bool bGeoTransform = false;
+    std::string osOutputSRS;
+    CPLStringList aosMetadata;
+    std::vector<double> adfBurnValues;
+    bool bQuiet = false;
+    bool bSetNoData = false;
+    std::string osNoData;
+    std::string osOutputFilename;
+    std::string osInputFilename;
+    std::string osFormat;
+    CPLStringList aosCreateOptions;
+};
+
+/************************************************************************/
+/*                   GDALCreateAppOptionsGetParser()                   */
+/************************************************************************/
 
+static std::unique_ptr<GDALArgumentParser>
+GDALCreateAppOptionsGetParser(GDALCreateOptions *psOptions)
 {
-    char *pszEnd = nullptr;
-    CPLStrtod(pszArg, &pszEnd);
-    return pszEnd != nullptr && pszEnd[0] == '\0';
+    auto argParser = std::make_unique<GDALArgumentParser>(
+        "gdal_create", /* bForBinary */ true);
+
+    argParser->add_description(
+        _("Create a raster file (without source dataset)."));
+
+    argParser->add_epilog(_(
+        "For more details, consult the full documentation for the gdal_create "
+        "utility: http://gdal.org/gdal_create.html"));
+
+    argParser->add_output_type_argument(psOptions->eDT);
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_argument("-outsize")
+        .metavar("<xsize> <ysize>")
+        .nargs(2)
+        .scan<'i', int>()
+        .action(
+            [psOptions](const std::string &s)
+            {
+                if (!psOptions->bPixelsSet)
+                {
+                    psOptions->nPixels = atoi(s.c_str());
+                    psOptions->bPixelsSet = true;
+                }
+                else
+                {
+                    psOptions->nLines = atoi(s.c_str());
+                }
+            })
+        .help(_("Set the size of the output file."));
+
+    argParser->add_argument("-bands")
+        .metavar("<count>")
+        .store_into(psOptions->nBandCount)
+        .help(_("Set the number of bands in the output file."));
+
+    argParser->add_argument("-burn")
+        .metavar("<value>")
+        .nargs(nargs_pattern::at_least_one)
```

Comment:
.append() should probably be added too, since the current synopsis indicates "[-burn <value>]..."

Testing a bit, I see we currently support all of 3 below variants:
```
gdal_create -outsize 1 1 -bands 3 -burn 1 2 3 out.tif
gdal_create -outsize 1 1 -bands 3 -burn "1 2 3" out.tif
gdal_create -outsize 1 1 -bands 3 -burn 1 -burn 2 -burn 3 out.tif
```
the later one is the only one actually documented (or at least this is my understanding of the synopsis and documentation), but the one tested per test_gdal_create_pdf_tif is `` -burn 1 2``
If that's not too inconvenient, we could try to still support the 3 variants?

Reply:
done